lms/djangoapps/learner_dashboard/tests/test_views.py

```diff
@@ -114,18 +120,19 @@ def test_api_returns_discussions_iframe(self):
         self.assertIsInstance(response.data['discussion']['iframe'], Markup)
         self.assertIn('iframe', str(response.data['discussion']['iframe']), )
 
-    def test_program_does_not_exist(self):
+    def test_program_empty_response(self):
         """
-        Test if API returns 404 in case program does not exist
-        """
-        response = self.client.get(reverse('program_discussion', kwargs={'program_uuid': str(uuid4())}))
-        self.assertEqual(response.status_code, 404)
-
-    def test_program_access_denied(self):
-        """
-        Test if API returns 403 in case user has no access to program
+        Test if API returns default response in case user has no access to program
```

Comment:
This docstring also needs to improve and explain the scenario under test. 

Test that the API does not break if the discussion provider is not configured, instead it should return empty data for discussion iframe.

Reply:
This is not the scenario that is being tested here. The current docstring is correct.

Reply:
1) "default response" is misleading here. 
2) please mention what kind of access.